doc/sphinx/language/gallina-specification-language.rst

```diff
@@ -139,50 +139,48 @@ presentation of Cic is given in Chapter :ref:`calculusofinductiveconstructions`.
 are given in Chapter :ref:`extensionsofgallina`. How to customize the syntax
 is described in Chapter :ref:`syntaxextensionsandinterpretationscopes`.
 
-.. insertgram term binders_opt
 
-.. productionlist:: coq
-   term : forall `open_binders` , `term`
-        : fun `open_binders` => `term`
-        : `term_let`
-        : if `term` `as_return_type_opt` then `term` else `term`
-        : `term_fix`
-        : `term100`
-   term100 : `term_cast`
-           : `term10`
-   term10 : `term1` `args`
-          : @ `qualid` `universe_annot_opt` `term1_list_opt`
-          : `term1`
-   args : `args` `arg`
-        : `arg`
-   arg : ( `ident` := `term` )
-       : `term1`
-   term1_list_opt : `term1_list_opt` `term1`
-                  : `empty`
-   empty :
-   term1 : `term_projection`
-         : `term0` % `ident`
-         : `term0`
-   args_opt : `args`
-            : `empty`
-   term0 : `qualid` `universe_annot_opt`
-         : `sort`
-         : `numeral`
-         : `string`
-         : _
-         : `term_evar`
-         : `term_match`
-         : ( `term` )
-         : {| `fields_def` |}
-         : `{ `term` }
-         : `( `term` )
-         : ltac : ( `ltac_expr` )
-   fields_def : `field_def` ; `fields_def`
-              : `field_def`
-              : `empty`
-   field_def : `qualid` `binders_opt` := `term`
-   binders_opt : `binders`
-               : `empty`
+.. prodn::
+   y ::= {*| {*| {*| {*| {*| {*| @tactic } } } } } }
+
+.. prodn::
+   x ::= under {? @r_prefix } @term {? => {+ @i_item}} {? do {| @tactic | [ {*| @tactic } ] } } under {? @r_prefix } @term {? => {+ @i_item}} {? do {| @tactic | [ {*| @tactic } ] } }
+
+.. prodn::
+   blorch ::= term term10
+
+.. prodn::
+   term ::= forall @open_binders , @term
+   += fun @open_binders => @term
+   += @term_let
+   += if @term @as_return_type_opt then @term else @term
+   += @term_fix
+   += @term100
+   term100 ::= @term_cast
+   += @term10
+   term10 ::= @term1 {+ @arg }
+   += @ @qualid @univ_instance {* @term1 }
+   += @term1
```

Comment:
I would have used `*` on the first line to eliminate the third.
And `univ_instance` is an option that could be inlined.

Reply:
I didn't spend a lot of time trying to perfect the details of the presentation--still work to do on the basics.

For inlining, I think we should consider a) whether the nonterminal is used in multiple distinct places in the grammar, b) whether it will be helpful to refer to the nonterminal in the text (e.g. to describe the use of the associated production(s) in one place rather than repeating) and c) whether the production will fit on a single line.

If we use multiline `prodn`s, then we should include `|` characters in the table so a multiline production won't be confused with two or more separate productions.  Below `x` is a multiline prodn, while `term` has only single-line productions.  I prefer to avoid multiline `prodn`s and not showing the `|` in the table--it seems cleaner but may still be ambiguous if the browser window is narrow, causing more lines to be split.  We can't control that.  What do you think?

We can still have multiline commands and tactics if we wish.

![image](https://user-images.githubusercontent.com/1253341/69753206-d5f0e900-1107-11ea-8396-9af60af60e74.png)


![image](https://user-images.githubusercontent.com/1253341/69753177-c1145580-1107-11ea-93ff-dedad773f3a2.png)

Reply:
It looks good with the `|` characters. Indeed, like you said, we can't have perfect control over what the browser window will show. However, we could also consider wrapping up without indentation to distinguish multi-line productions from multiple single-line ones.

Reply:
> we could also consider wrapping up without indentation to distinguish multi-line productions from multiple single-line ones

Meaning the first line of the production is not indented and subsequent (continutation) lines are?

Reply:
No, the converse.

Reply:
Still no idea what you have in mind.  You're suggesting something like one of these?  (Only #1 makes sense to me.)

```
# 1
foo     := ident blah blah blah blah blah blah blah blah blah blah blah blah blah blah
               continued line blah blah blah 

# 2
foo     :=     ident blah blah blah blah blah blah blah blah blah blah blah blah blah blah
           continued line blah blah blah 

# 3
foo     := ident blah blah blah blah blah blah blah blah blah blah blah blah blah blah
continued line blah blah blah 
```

Reply:
I'm suggesting `#3` but that was just an idea in passing. You don't have to adopt it if you don't like it.

Reply:
I see.  #3 would be difficult.  I'm using a 3 column table so things line up vertically.  AFAIK there's no easy way to show the continuation of the third column across all three columns.